rust/sbd-client/src/send_buf.rs

```diff
@@ -0,0 +1,159 @@
+use super::*;
+
+use std::collections::VecDeque;
+
+pub struct SendBuf {
+    pub ws: raw_client::WsRawSend,
+    pub buf: VecDeque<Vec<u8>>,
+    pub out_buffer_size: usize,
+    pub origin: tokio::time::Instant,
+    pub limit_rate: u64,
+    pub idle_keepalive_nanos: u64,
+    pub next_send_at: u64,
+    pub last_send: u64,
+}
+
+impl SendBuf {
+    /// construct a new send buf
+    pub fn new(
+        ws: raw_client::WsRawSend,
+        out_buffer_size: usize,
+        limit_rate: u64,
+        idle_keepalive: std::time::Duration,
+        pre_sent_bytes: usize,
+    ) -> Self {
+        let mut this = Self {
+            ws,
+            buf: VecDeque::default(),
+            out_buffer_size,
+            origin: tokio::time::Instant::now(),
+            limit_rate,
+            idle_keepalive_nanos: idle_keepalive.as_nanos() as u64,
+            next_send_at: 0,
+            last_send: 0,
+        };
+
+        let now = this.origin.elapsed().as_nanos() as u64;
+
+        this.next_send_at = std::cmp::max(now, this.next_send_at)
```

Comment:
`this.next_send_at` is initialized as 0 above so the `cmp::max` is not required. this had me 2nd guess for a second

Reply:
Done: https://github.com/holochain/sbd/pull/2/commits/f1bad2fa0a69fa1623f15de298ea91d43a354d52